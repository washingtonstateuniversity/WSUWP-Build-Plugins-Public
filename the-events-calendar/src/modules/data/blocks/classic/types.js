/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_INITIAL_STATE = `${ PREFIX_EVENTS_STORE }/CLASSIC_BLOCK/SET_INITIAL_STATE`;

export const SET_CLASSIC_DETAILS_TITLE = `${ PREFIX_EVENTS_STORE }/SET_CLASSIC_DETAILS_TITLE`;
export const SET_CLASSIC_ORGANIZERS_TITLE = `${ PREFIX_EVENTS_STORE }/SET_CLASSIC_ORGANIZERS_TITLE`;
