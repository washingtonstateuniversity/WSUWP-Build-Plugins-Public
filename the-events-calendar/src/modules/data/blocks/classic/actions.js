/**
 * Internal dependencies
 */
import * as types from './types';

export const setDetailsTitle = ( title ) => ( {
	type: types.SET_CLASSIC_DETAILS_TITLE,
	payload: {
		title,
	},
} );

export const setOrganizerTitle = ( title ) => ( {
	type: types.SET_CLASSIC_ORGANIZERS_TITLE,
	payload: {
		title,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );
