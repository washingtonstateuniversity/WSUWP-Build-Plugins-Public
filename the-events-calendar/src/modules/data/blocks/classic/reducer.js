/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	detailsTitle: '',
	organizerTitle: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_CLASSIC_DETAILS_TITLE:
			return {
				...state,
				detailsTitle: action.payload.title,
			};

		case types.SET_CLASSIC_ORGANIZERS_TITLE:
			return {
				...state,
				organizerTitle: action.payload.title,
			};

		default:
			return state;
	}
};
