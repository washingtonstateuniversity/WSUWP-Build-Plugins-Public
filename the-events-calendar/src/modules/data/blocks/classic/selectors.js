/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const classicSelector = ( state ) => state.events.blocks.classic;

export const detailsTitleSelector = createSelector(
	[ classicSelector ],
	( values ) => values.detailsTitle,
);

export const organizerTitleSelector = createSelector(
	[ classicSelector ],
	( values ) => values.organizerTitle,
);
