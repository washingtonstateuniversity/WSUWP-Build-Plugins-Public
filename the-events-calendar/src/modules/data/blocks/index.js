/**
 * Internal dependencies
 */
import * as datetime from './datetime';

export { datetime };
export { default } from './reducer';
