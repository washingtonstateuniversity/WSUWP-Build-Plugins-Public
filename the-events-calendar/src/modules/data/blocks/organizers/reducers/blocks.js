/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import * as types from './../types';
import block from './block';

export const byId = ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
		case types.REMOVE_ORGANIZER_BLOCK:
			return {
				...state,
				[ action.payload.id ]: block( state[ action.payload.id ], action ),
			};
		default:
			return state;
	}
};

export const allIds = ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_ORGANIZER_BLOCK:
			return [ ...state, action.payload.organizer ];
		case types.REMOVE_ORGANIZER_BLOCK:
			return [ ...state ].filter( ( organizer ) => organizer !== action.payload.organizer );
		default:
			return state;
	}
};

export default combineReducers( {
	byId,
	allIds,
} );
