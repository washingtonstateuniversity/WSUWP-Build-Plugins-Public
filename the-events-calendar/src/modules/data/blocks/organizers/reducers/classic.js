import { uniq } from 'lodash';
/**
 * Internal dependencies
 */
import * as types from './../types';

export default ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_CLASSIC_ORGANIZERS:
			return uniq( [ ...state, action.payload.organizer ] );
		case types.REMOVE_CLASSIC_ORGANIZERS:
			return state.filter( ( organizer ) => organizer !== action.payload.id );
		case types.SET_CLASSIC_ORGANIZERS:
			return [ ...action.payload.organizers ];
		default:
			return state;
	}
};
