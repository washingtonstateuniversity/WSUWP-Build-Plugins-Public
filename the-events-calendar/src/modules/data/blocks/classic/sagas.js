/**
 * External Dependencies
 */
import { put, all, takeEvery } from 'redux-saga/effects';

/**
 * Internal dependencies
 */
import * as types from './types';
import { DEFAULT_STATE } from './reducer';
import * as actions from './actions';

export function* setInitialState( action ) {
	const { get } = action.payload;

	yield all( [
		put( actions.setDetailsTitle( get( 'detailsTitle', DEFAULT_STATE.detailsTitle ) ) ),
		put( actions.setOrganizerTitle( get( 'organizerTitle', DEFAULT_STATE.organizerTitle ) ) ),
	] );
}

export default function* watchers() {
	yield takeEvery( types.SET_INITIAL_STATE, setInitialState );
}
