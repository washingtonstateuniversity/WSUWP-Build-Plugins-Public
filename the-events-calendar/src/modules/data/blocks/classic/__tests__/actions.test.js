/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/events/data/blocks/classic';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Classic actions', () => {
	it( 'Should set initial state', () => {
		expect( actions.setInitialState( {} ) ).toMatchSnapshot();
	} );

	test( 'Action to set the organizer title', () => {
		expect( actions.setOrganizerTitle( 'Modern Tribe' ) ).toMatchSnapshot();
	} );

	test( 'Action to set the details title', () => {
		expect( actions.setDetailsTitle( 'Events' ) ).toMatchSnapshot();
	} );
} );
