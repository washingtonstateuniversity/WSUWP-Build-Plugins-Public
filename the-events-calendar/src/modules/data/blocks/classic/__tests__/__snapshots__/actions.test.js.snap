// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Classic actions Action to set the details title 1`] = `
Object {
  "payload": Object {
    "title": "Events",
  },
  "type": "@@MT/EVENTS/SET_CLASSIC_DETAILS_TITLE",
}
`;

exports[`[STORE] - Classic actions Action to set the organizer title 1`] = `
Object {
  "payload": Object {
    "title": "Modern Tribe",
  },
  "type": "@@MT/EVENTS/SET_CLASSIC_ORGANIZERS_TITLE",
}
`;

exports[`[STORE] - Classic actions Should set initial state 1`] = `
Object {
  "payload": Object {},
  "type": "@@MT/EVENTS/CLASSIC_BLOCK/SET_INITIAL_STATE",
}
`;
